### ISSUE TITLE
Signal forwarding

### ISSUE THREAD

**Issue by quantum5** [DMOJ/judge#186]
In `cptbox`, the debugged child process is suspended on signals, per normal `ptrace` behaviour. However, `cptbox` does not resend the signal to the child, and hence its signal handlers are never invoked. This behaviour was fine in all the years we have used the judge, but it might be the cause of some sketchiness.

However, the failure to implement #16 seems to be that Racket handles `SIGSEGV` (it apparently uses the signal in its normal operation), while we simply kill the child on `SIGSEGV` and call it a day. We may instead choose to forward `SIGSEGV` to the child's handler, but this may cause `RTE`s to turn into `IR`s. The consequences of such a change must be thoroughly explored before it is made.

**Comment by Xyene** [comment]
On one hand, this would allow us to provide better feedback for things like `std::bad_alloc` since it would printed as such in stderr, however in the short term this isn't really necessary. If we were to have a flag to enable signal forwarding to the child, we'd be able to use it for languages that actually require it, like Racket.

In such cases it's OK if there are no RTEs, and only IRs. Converting old native executors to be aware of this sounds like a not-worthwhile pain, but getting languages like Racket to work in the future is a potential benefit &mdash; even if they'll never RTE (is Racket even supposed to RTE normally? If it's anything like Java in terms of runtime it's fine if there are only IRs.) By making it optional, we can have both until &mdash; and if &mdash; we want to parse child stderr for feedback.

**Comment by quantum5** [comment]
For at least C and the like, libc still causes the program to exit with `SIGSEGV`, as the following session proves after 1c4aac75c6ed5acfa1ae3c983b45e2d9bdc0c1de:

```c
dmoj> submit helloworld C
int main() {
    int *i = 0;
    *i = 0;
}
Start grading helloworld/1 in C...
Test case  1 RTE [0.002s | 892kb] (Segmentation fault) {WA}
Done grading helloworld/1.
```

**Comment by quantum5** [comment]
This also fixes #179:

```cpp
dmoj> submit helloworld CPP11
#include <new>

int main() {
    throw std::bad_alloc();
}
Start grading helloworld/1 in CPP11...
Test case  1 RTE [0.004s | 1344kb] (Aborted) {WA}
Done grading helloworld/1.
```

**Comment by quantum5** [comment]
Signals are correctly forwarded now.
